### PR TITLE
Update to map NIST tags from CCI refs

### LINKED ISSUE #54: nessus_mapper CCI to NIST Mapping
`nessus_mapper` should map CCI numbers if available to NIST 800-53 references.

### PR DESCRIPTION
**Pull request by rx294**
nessus_mapper should map CCI numbers if available to NIST 800-53 references.

Fixes #54

### CODE REVIEW

**Comment by ejaronne** [comment]
For Nessus compliance scans, I recommend seeking CCI references first, then 800-53 if CCIs aren't available. Reason: I observed that for STIG-based Nessus compliance scans, all checks have CCIs, but not all have 800-53 references. Also and more importantly, the 800-53 references don't match correct CCI-to-800-53 mappings from DISA's CCI list.

**Comment by aaronlippold** [comment]
@rx294 please add unit and functional tests

File: lib/heimdall_tools/nessus_mapper.rb [hunk 1]
```diff
@@ -107,7 +120,29 @@ def finding(issue, timestamp)
...
+    def cci_nist_tag(cci_refs)
+      nist_tags = []
+      cci_refs.each do | cci_ref |
+        item_node = @cci_xml.xpath("//cci_list/cci_items/cci_item[@id='#{cci_ref}']")[0] unless @cci_xml.nil?
+        unless item_node.nil?
+          nist_ref = item_node.xpath('./references/reference[not(@version <= preceding-sibling::reference/@version) and not(@version <=following-sibling::reference/@version)]/@index').text
+          nist_ver = item_node.xpath('./references/reference[not(@version <= preceding-sibling::reference/@version) and not(@version <=following-sibling::reference/@version)]/@version').text
+        end
+        nist_tags << nist_ref
+        nist_tags << "Rev_#{nist_ver}"
```
Comment (aaronlippold) [inline lib/heimdall_tools/nessus_mapper.rb:140]: Please remove the rev #

File: lib/heimdall_tools/nessus_mapper.rb [hunk 2]
```diff
@@ -2,10 +2,12 @@
 require 'csv'
 require 'heimdall_tools/hdf'
 require 'utilities/xml_to_hash'
+require 'nokogiri'
 
 RESOURCE_DIR = Pathname.new(__FILE__).join('../../data')
```
Comment (aaronlippold) [inline lib/heimdall_tools/nessus_mapper.rb:7]: There was some conversation with @rbclark and @Bialogs on the right way to do this at the gem level to make sure we are always referencing the data files in the right location and that the data in the compiled gem was not actually an xls but a proper ruby data object. Does this incorporate this?

File: lib/heimdall_tools/nessus_mapper.rb [hunk 3]
```diff
@@ -107,7 +120,29 @@ def finding(issue, timestamp)
       [finding]
     end
 
-    def nist_tag(pluginfamily, pluginid)
+    def read_cci_xml
+      cci_list_path = File.join(File.dirname(__FILE__), '../data/U_CCI_List.xml')
```
Comment (aaronlippold) [inline lib/heimdall_tools/nessus_mapper.rb:124]: See other comment: @rbclark and @Bialogs - does this work to ensure that we will always have access to the data file even when we gem install?

File: lib/heimdall_tools/nessus_mapper.rb (hunk 2 above)
Comment (rx294) [reply lib/heimdall_tools/nessus_mapper.rb:7]: Yes Pathname gets us the absolute path and is suitable to work from a gem.

**Comment by rx294** [comment]
> * [ ]  Please add at least one functional and unit test with our test data in the 'gem installed' state and the 'from code' state
> * [ ]  Plese review the changes in the PRs for Heimdall Tools and InSpec Tools to ensure we are using the correct data file access and 'convertion to ruby object in the compiled state' to make sure we don't have file load issues again

Functional test performed locally;
Unit tests will be added on a later PR

**Pull request merged**